run.go

```diff
@@ -84,12 +86,32 @@ func TestingT(testingT *testing.T) {
 // provided run configuration.
 func RunAll(runConf *RunConf) *Result {
 	result := Result{}
+	if !CustomParallelSuiteFlag {
+		for _, suite := range allSuites {
+			result.Add(Run(suite, runConf))
+		}
+		return &result
+	}
+
+	wg := sync.WaitGroup{}
+	suiteRunners := make([]*suiteRunner, 0, len(allSuites))
 	for _, suite := range allSuites {
-		result.Add(Run(suite, runConf))
+		suiteRunners = append(suiteRunners, parallelRun(suite, runConf, &wg))
```

Comment:
Why add a new function `parallelRun()` instead of reusing the old logic?

Reply:
We need to execute `setUpSuite` serially first. And I hope not to affect the original function.

Reply:
But this can also be used with the original.